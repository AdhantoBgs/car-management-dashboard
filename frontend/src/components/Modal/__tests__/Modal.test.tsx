import React from "react";
import { render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import { BrowserRouter as Router } from "react-router-dom";
import ModalBtn from "..";

describe("<ModalBtn />", () => {
	test("renders correctly", () => {
		const {asFragment} = render(
			<Router>
				<ModalBtn />
			</Router>
		);
        
		expect(asFragment()).toMatchSnapshot();
	});

	test("shows a modal when delete button is clicked", () => {
		const {getByText} = render(
			<Router>
				<ModalBtn />
			</Router>
		);

		fireEvent.click(getByText(/Delete/i));
		expect(getByText(/Menghapus Data Mobil/i)).toBeInTheDocument();
	});

	test("redirects to the update page when the edit button is clicked", () => {
		const {getByText} = render(
			<Router>
				<ModalBtn />
			</Router>
		);

		fireEvent.click(getByText(/Edit/i));
		expect(() => getByText(/Menghapus Data Mobil/i)).toThrow();
	});
});