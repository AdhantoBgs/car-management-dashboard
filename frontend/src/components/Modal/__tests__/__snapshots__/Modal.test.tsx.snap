// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModalBtn /> renders correctly 1`] = `
<DocumentFragment>
  <div>
    <div
      class="btn-content ant-flex css-dev-only-do-not-override-6j9yrn"
    >
      <button
        class="ant-btn css-dev-only-do-not-override-6j9yrn ant-btn-default"
        style="width: 100%; font-size: 14px; color: rgb(250, 44, 90); background: white; border: 1px solid #fa2c5a; border-radius: 3px; padding: .5rem .75rem; margin-right: .5rem;"
        type="button"
      >
        <span
          aria-label="delete"
          class="anticon anticon-delete"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="delete"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M360 184h-8c4.4 0 8-3.6 8-8v8h304v-8c0 4.4 3.6 8 8 8h-8v72h72v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80h72v-72zm504 72H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zM731.3 840H292.7l-24.2-512h487l-24.2 512z"
            />
          </svg>
        </span>
        <span>
          Delete
        </span>
      </button>
      <button
        class="ant-btn css-dev-only-do-not-override-6j9yrn ant-btn-default"
        style="width: 100%; font-size: 14px; font-weight: bold; color: white; background: rgb(92, 184, 95); border: 1px solid #5cb85f; border-radius: 3px; padding: .5rem .75rem; margin-left: .5rem;"
        type="button"
      >
        <span
          aria-label="form"
          class="anticon anticon-form"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="form"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M904 512h-56c-4.4 0-8 3.6-8 8v320H184V184h320c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V520c0-4.4-3.6-8-8-8z"
            />
            <path
              d="M355.9 534.9L354 653.8c-.1 8.9 7.1 16.2 16 16.2h.4l118-2.9c2-.1 4-.9 5.4-2.3l415.9-415c3.1-3.1 3.1-8.2 0-11.3L785.4 114.3c-1.6-1.6-3.6-2.3-5.7-2.3s-4.1.8-5.7 2.3l-415.8 415a8.3 8.3 0 00-2.3 5.6zm63.5 23.6L779.7 199l45.2 45.1-360.5 359.7-45.7 1.1.7-46.4z"
            />
          </svg>
        </span>
        <span>
          Edit
        </span>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
